common/src/com/thoughtworks/go/domain/MaterialRevisions.java

```diff
@@ -272,12 +272,12 @@ private MaterialRevisions filter(MaterialRevisions other) {
         return filtered;
     }
 
-    public Map<String, String> getNamedRevisions() {
-        Map<String, String> results = new HashMap<String, String>();
+    public Map<CaseInsensitiveString, String> getNamedRevisions() {
+        Map<CaseInsensitiveString, String> results = new HashMap<CaseInsensitiveString, String>();
```

Comment:
Changing the return type allows for case-insensitive lookup -- which is required.